packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -544,6 +546,7 @@ class MongoshNodeRepl implements EvaluationListener {
     });
 
     repl.on('exit', () => {
+      if (this._runtimeState) this._runtimeState.repl = null;
```

Comment:
what are the implications of this being set to null? is this to force a quicker garbage collection / flushing? might be worth leaving a comment, though perhaps I could just be lacking context

Reply:
Oh yeah, comment is a good idea – also added in 3bcdc5295025302154e06fa5f746f150aa5ca17e. (The shell can also exit when stdin ends, which would result in us calling this listener here first, before we call `.onExit()` through some other means)